Static type hint error on class pandera DataFrame

manel-ab opened the pull request:
I stated an error referencing this issue: [https://github.com/unionai-oss/pandera/issues/1205](url).
As commented by @cosmicBboy, here is my solution.

cosmicBboy commented:
thanks! sorry for this bit of process, but could you follow the steps [here](https://pandera.readthedocs.io/en/stable/CONTRIBUTING.html#dco-signing-commits) to get the DCO check to pass?

manel-ab commented:
> > Done! I also added another commit to use the schema paratemer instead of the generic type
> 
> mypy doesn't seem to like this: https://github.com/unionai-oss/pandera/actions/runs/5134262937/jobs/9239769839?pr=1207
> 
> Feel free to `type: ignore` this if you can't find a way to make this work the right way.

Hi, I have been trying to solve the mypy problem but I cannot come up with anything. It is weird, in pyright works: https://pyright-playground.decorator-factory.su/?gzip=H4sIAGdJeGQC_42QTQvCMAyG7_0VYacWdOCOA8We_APFy5ASZqaFrRttFPz37sup4MEcwkt48-SjCm0D_Oicv4BrujYwmEdHqzEfMazgQJ6CK6eKrh1GIQxsXwaZmESJssYYQcvZXJiTygX0caYKrHXesbUyUl3N9SHSNF30KPaRkV3ZEF_b89LOFHkjMR8nDmRY7yDRvUrerEB8Cx50gSep_qVmf1PNFxX783U67eU8KxHoTljb_o8kUYkPQ_bL8AQOuJLBdQEAAA%3D%3D

whereas in mypy does not: https://mypy-play.net/?mypy=latest&python=3.11&flags=verbose&gist=3cb8319b768b1be2e91fbf77347c7a57

So I just added the type:ignore as commented by you.

cosmicBboy commented:
thanks, and congrats on your first contribution to pandera @manel-ab ! 🚀

manel-ab commented:
> thanks, and congrats on your first contribution to pandera @manel-ab ! rocket

thank you!
